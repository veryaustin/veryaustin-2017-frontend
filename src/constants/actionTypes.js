export const DATA_FETCH = 'DATA_FETCH';