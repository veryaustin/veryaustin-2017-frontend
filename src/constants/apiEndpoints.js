export const DATA_ROOT_URL = "https://veryaustin-55602.firebaseio.com";
export const STATIC_ASSETS_URL = "http://static.veryaustin.com";
