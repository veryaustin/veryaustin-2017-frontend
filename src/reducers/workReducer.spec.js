import * as types from '../constants/actionTypes';
import reducer from './workReducer';

describe('Work Reducer',() => {
  it('should return the initial state', () => {
    const action = {type: 'unknown'};
    expect(reducer(undefined, action)).toMatchSnapshot();
  });

  it('should handle the WORK_FETCH action', () => {
    const initialState = {};

    const testData = {
      title: 'Work',
      slogan: 'Test Slogan',
      projects: [
        {
          title: 'Test Project 1 Title',
          summary: 'Test Project 1 Summary',
          description: 'This is a description for Test Project 1',
          image_url: 'http://test_project_1_image_url.com',
          repo_url: 'https://test_project_1_repo_url.com',
          site_url: 'http://test_project_1_site_url.com'
        },
        {
          title: 'Test Project 2 Title',
          summary: 'Test Project 2 Summary',
          description: 'This is a description for Test Project 2',
          image_url: 'http://test_project_2_image_url.com',
          repo_url: 'https://test_project_2_repo_url.com',
          site_url: 'http://test_project_2_site_url.com'
        }
      ]
    };

    const action = {
      type: types.DATA_FETCH,
      payload: {data: testData}
    };
    
    expect(reducer(initialState, action)).toMatchSnapshot();
  });
});