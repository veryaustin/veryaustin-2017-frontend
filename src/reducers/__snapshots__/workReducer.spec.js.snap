exports[`Work Reducer should handle the WORK_FETCH action 1`] = `Object {}`;

exports[`Work Reducer should return the initial state 1`] = `Object {}`;
